persist/fs/commitlog/writer.go

```diff
@@ -0,0 +1,281 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package commitlog
+
+import (
+	"bufio"
+	"encoding/binary"
+	"errors"
+	"hash/adler32"
+	"io"
+	"os"
+	"time"
+
+	"github.com/m3db/m3db/generated/proto/schema"
+	"github.com/m3db/m3db/interfaces/m3db"
+	"github.com/m3db/m3db/persist/fs"
+	"github.com/m3db/m3x/time"
+
+	"github.com/golang/protobuf/proto"
+)
+
+const (
+	// The length to reserve for a chunk header:
+	// - size uint32
+	// - checksumSize uint32
+	// - checksumData uint32
+	chunkReserveHeaderLen = 4 + 4 + 4
```

Comment:
Worth having consts for size, checksumSize, checksumData rather than have this in a comment. I'm sure checksumSize is used elsewhere.